### PR TITLE
Update schemas according to optimade v0.15.3

### PR DESCRIPTION
**Pull request by CasperWA**
This removes:
- HTTPValidationError
- ValidationError

And adds proper support for response codes and their response models for
each of the API routes.

It also updates a couple of descriptions by removing end-of-the-line
white space.

It also updates the descriptions according to the latest changes in the specification, specifically the changes to the `structures` entry resource.

### CODE REVIEW

**Comment by ml-evs** [comment]
So the description tweaks introduced in #349 are not in our models yet (and thus not in this schema). I've just made https://github.com/Materials-Consortia/optimade-python-tools/pull/847 to fix this, we may need to release 0.15.3 before merging this...

**Review by merkys — APPROVED** [review]
I gave it a quick look, and it seems OK. Most of the changes indeed seem to be connected with HTTP error descriptions.

I admit at first I was confused to read "optimade v0.15.2" as we did not even had such release of the spec :smile:

**Review by ml-evs — CHANGES_REQUESTED** [review]
Just doing this so we don't accidentally merge early :grin:

**Comment by CasperWA** [comment]
> So the description tweaks introduced in #349 are not in our models yet (and thus not in this schema). I've just made [Materials-Consortia/optimade-python-tools#847](https://github.com/Materials-Consortia/optimade-python-tools/pull/847) to fix this, we may need to release 0.15.3 before merging this...

Sorry, I might have misunderstood this from our talk yesterday. But just ping me when you need reviews 👍

**Comment by CasperWA** [comment]
> > So the description tweaks introduced in #349 are not in our models yet (and thus not in this schema). I've just made [Materials-Consortia/optimade-python-tools#847](https://github.com/Materials-Consortia/optimade-python-tools/pull/847) to fix this, we may need to release 0.15.3 before merging this...
> 
> Sorry, I might have misunderstood this from our talk yesterday. But just ping me when you need reviews 👍 And you can push to this branch to update the schemas further afterwards.

I've updated it all myself now 😅 

The newest commit (7f78481) comes from the newly released OPTIMADE Python tools v0.15.3 after retrieving and updating to the latest descriptions from the specification in this repository for the models, specifically the `structures` entry resource.

File: schemas/index_openapi_schema.json [hunk 1]
```diff
@@ -2,8 +2,8 @@
   "openapi": "3.0.2",
   "info": {
     "title": "OPTIMADE API - Index meta-database",
-    "description": "The [Open Databases Integration for Materials Design (OPTIMADE) consortium](https://www.optimade.org/) aims to make materials databases interoperational by developing a common REST API.\nThis is the \"special\" index meta-database.\n\nThis specification is generated using [`optimade-python-tools`](https://github.com/Materials-Consortia/optimade-python-tools/tree/v0.13.3) v0.13.3.",
-    "version": "1.0.0~develop"
+    "description": "The [Open Databases Integration for Materials Design (OPTIMADE) consortium](https://www.optimade.org/) aims to make materials databases interoperational by developing a common REST API.\nThis is the \"special\" index meta-database.\n\nThis specification is generated using [`optimade-python-tools`](https://github.com/Materials-Consortia/optimade-python-tools/tree/v0.15.3) v0.15.3.",
+    "version": "1.0.1"
```
Comment (ml-evs) [inline schemas/index_openapi_schema.json:6]: Should we wait to bump this?
```suggestion
    "version": "1.0.0~develop"
```

Comment (CasperWA) [reply schemas/index_openapi_schema.json:6]: Ah right, sorry - changed that in the other one, not here.

File: schemas/openapi_schema.json [hunk 2]
```diff
@@ -2,8 +2,8 @@
   "openapi": "3.0.2",
   "info": {
     "title": "OPTIMADE API",
-    "description": "The [Open Databases Integration for Materials Design (OPTIMADE) consortium](https://www.optimade.org/) aims to make materials databases interoperational by developing a common REST API.\n\nThis specification is generated using [`optimade-python-tools`](https://github.com/Materials-Consortia/optimade-python-tools/tree/v0.13.3) v0.13.3.",
-    "version": "1.0.0~develop"
+    "description": "The [Open Databases Integration for Materials Design (OPTIMADE) consortium](https://www.optimade.org/) aims to make materials databases interoperational by developing a common REST API.\n\nThis specification is generated using [`optimade-python-tools`](https://github.com/Materials-Consortia/optimade-python-tools/tree/v0.15.3) v0.15.3.",
+    "version": "1.0.1"
```
Comment (CasperWA) [inline schemas/openapi_schema.json:6]:
```suggestion
    "version": "1.0.0~develop"
```

**Review by ml-evs — APPROVED** [review]

**Review by merkys — APPROVED** [review]
Glad to see the effect of #349 also in here :)

**Pull request merged**